### PR TITLE
Thumbnail Gallery Component

### PR DESCRIPTION
**Pull request by prodan-eric**
https://opentechdev.atlassian.net/jira/software/projects/DLCL/boards/17?assignee=637212f85fc160544e16fa5f&selectedIssue=DLCL-789

### CODE REVIEW

File: src/stories/DlThumbnailGallery.stories.js
Comment (fadiDL) [inline src/stories/DlThumbnailGallery.stories.js:1]: Please refactor this to the new style of stories and documentation with jsdoc inside the component itself.
checkout `DlEllipsis.stories.ts` for reference and `DlEllipsis.vue` for the doc

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: src/components/compound/DlThumbnailGallery/DlThumbnailGallery.vue
[the remark this answers is not in the document]
Comment (prodan-eric) [reply src/components/compound/DlThumbnailGallery/DlThumbnailGallery.vue:1]: Clicking an image directly updates the v-model, so the "selected" event would be equivalent to listening to @update:modelValue from outside

File: src/components/compound/DlThumbnailGallery/index.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import DlThumbnailGallery from './DlThumbnailGallery.vue'
+import { DlThumbnail } from './types'
+export { DlThumbnailGallery }
+export type { DlThumbnail }
```
[the remark this answers is not in the document]
Comment (prodan-eric) [reply src/components/compound/DlThumbnailGallery/index.ts:4]: I removed it and added it separately in the demo file for demo purpose

File: src/demos/index.ts [hunk 2]
```diff
@@ -108,9 +109,66 @@ export default {
     DlBarChartDemo,
     DlLineChartDemo,
     DlSpinner,
-    DlConfusionMatrix,
-    DlToggleButtonDemo,
     DlEllipsisDemo,
+    DlThumbnailGallery
+}
+
+export const demoState = {
```
Comment (fadiDL) [inline src/demos/index.ts:116]: We removed demoState in change for a export default a long time ago. please revert this to match.

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: src/demos/index.ts (hunk 2 above)
Comment (fadiDL) [reply src/demos/index.ts:116]: I solved it

File: src/components/essential/DlIcon/DlIcon.vue
Comment (fadiDL) [inline src/components/essential/DlIcon/DlIcon.vue:1]: This is a merge conflict. this code should not be reverted!

Comment (fadiDL) [reply src/components/essential/DlIcon/DlIcon.vue:1]: fixing conflicts

**Review by fadiDL — APPROVED** [review]

**Comment by fadiDL** [comment]
@prodan-eric please pay attention in the future to incorrect merge conflict resolutions. this branch contained changess that revert/ignore fixes we already introduced.

**Pull request merged**